### PR TITLE
Use structlog for mapper logging

### PR DESCRIPTION
**Pull request by rra**
Rather than initializing the root logger, change SlackGitHubMapper
to take an optional logger and get a logger from structlog to pass
in.  This will honor the logging settings for the application
without affecting other users of the root logger, such as the
slack module (which is very chatty at DEBUG).

### CODE REVIEW

File: src/checkerboard/app.py [hunk 1]
```diff
@@ -106,7 +101,8 @@
     config : `Configuration`
         The Checkerboard application configuration.
     """
-    mapper = SlackGitHubMapper(slack, config.profile_field)
+    logger = get_logger(None)
```
Comment (jonathansick) [inline src/checkerboard/app.py]: what does `get_logger(None)` do?

I'm familiar with `get_logger(__name__)` or `get_logger(config.logger_name)`.

Comment (rra) [reply src/checkerboard/app.py]: I'm trying to use the root logger per the recommendation in the Safir docs. Is that how to do it? That's how to get the root logger from `logging`, but I've not used `structlog` before.

Comment (jonathansick) [reply src/checkerboard/app.py]: Oh my bad 🤦‍♂ I didn't mean _root_ logger, I meant that Safir by default configures the logger for the application's module name (well, technically it's configured via `config.logger_name`, which the template sets to the module name.)

In other words, Safir configures the `checkerboard` logger, so any loggers using that namespace or a subnamespace like `checkerboard.mapper` or `checkerboard.app` would use that logger configuration.

I need to update those docs to be more careful about the word "root"!

Comment (rra) [reply src/checkerboard/app.py]: Ah, thank you. Now I understand what's going on.

**Pull request merged**